dts/bindings/base/mem-region.yaml

```diff
@@ -11,3 +14,10 @@ properties:
       memory region in the final executable. The region address and size
       is taken from the <reg> property, while the name is the value of
       this property.
+
+  zephyr,memory-region-mpu:
+    type: string
```

Comment:
note: could likely be an `int`, and a header with valid attributes be provided (each SoC could provide their own set if necessary)

Reply:
If it has no impact on code size, I would prefer to use string, at least in device tree. The reason is just simple; better readability.

Reply:
2 votes here for the strings.